### PR TITLE
Patch tests after deprecation

### PR DESCRIPTION
**Pull request by AmintorDusko**
**Context:**
Patching stable lightning tests for the latest pennylane.

[stable/latest](https://github.com/PennyLaneAI/pennylane-lightning/actions/runs/10965680477)

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

### CODE REVIEW

File: .github/workflows/tests_lqcpu_python.yml [hunk 1]
```diff
@@ -195,6 +195,13 @@ jobs:
           git log -1 --format='%H'
           git status
 
+      - name: Patching tests following deprecation (remove after v0.39)
```
Comment (maliasadi) [inline .github/workflows/tests_lqcpu_python.yml:198]:
```suggestion
      - name: Patching tests following deprecation (FIXME: remove after v0.39)
```

File: .github/workflows/tests_lqcpu_python.yml [hunk 2]
```diff
@@ -195,6 +195,13 @@ jobs:
...
+        if: inputs.lightning-version == 'stable' && inputs.pennylane-version != 'stable'
+        run: |
+          cd tests
```
Comment (maliasadi) [inline .github/workflows/tests_lqcpu_python.yml:201]:
```suggestion
          pushd tests
```

File: .github/workflows/tests_lqcpu_python.yml [hunk 3]
```diff
@@ -195,6 +195,13 @@ jobs:
...
+          grep -rl "default.qubit.legacy" . | xargs sed -i "s|default.qubit.legacy|default.qubit|g";
+          cd ..
```
Comment (maliasadi) [inline .github/workflows/tests_lqcpu_python.yml:203]:
```suggestion
          popd
```

File: .github/workflows/tests_lqcpu_python.yml [hunk 4]
```diff
@@ -195,6 +195,13 @@ jobs:
           git log -1 --format='%H'
           git status
 
+      - name: Patching tests following deprecation (remove after v0.39)
+        if: inputs.lightning-version == 'stable' && inputs.pennylane-version != 'stable'
```
Comment (maliasadi) [inline .github/workflows/tests_lqcpu_python.yml:199]: Shouldn't pennylane version be 'latest' as we're trying to fix the issue with stable/latest?

**Review by maliasadi — COMMENTED** [review]
Thanks @AmintorDusko! 🦅

**Review by multiphaseCFD — APPROVED** [review]
LGTM! Thanks @AmintorDusko .

File: .github/workflows/tests_lqcpu_python.yml (hunk 4 above)
Comment (AmintorDusko) [reply .github/workflows/tests_lqcpu_python.yml:199]: I didn't have much time to think about that.
This non-pervasive patch only does something if something needs to be done.
I can change it if this is important for you, though.
Just let me know. 🙂

Comment (maliasadi) [reply .github/workflows/tests_lqcpu_python.yml:199]: I only wonder about the correctness of this condition. We need to ensure that it won't break other checks in the plugin-matrix. If you're confident about it and have already fully checked the plugin-matrix, I won't have any concern with this as we will remove it in the next release.

Comment (maliasadi) [reply .github/workflows/tests_lqcpu_python.yml:199]: Do you know why we still get lgpu file-name errors in stable/stable? 
- https://github.com/PennyLaneAI/pennylane-lightning/actions/runs/10998801618
- https://github.com/PennyLaneAI/pennylane-lightning/actions/runs/10998790742

File: .github/workflows/tests_lqcpu_python.yml [hunk 5]
```diff
@@ -195,6 +195,13 @@ jobs:
           git log -1 --format='%H'
           git status
 
+      - name: Patching tests following deprecation (FIXME: remove after v0.39)
+        if: inputs.lightning-version == 'stable' && inputs.pennylane-version != 'stable'
```
Comment (AmintorDusko) [inline .github/workflows/tests_lqcpu_python.yml:199]:
```suggestion
        if: inputs.lightning-version == 'stable' && inputs.pennylane-version == 'stable'
```

File: .github/workflows/tests_lqcpu_python.yml (hunk 4 above)
Comment (AmintorDusko) [reply .github/workflows/tests_lqcpu_python.yml]: I was avoiding triggering the CIs without need. But I have no problem changing.

Comment (AmintorDusko) [reply .github/workflows/tests_lqcpu_python.yml]: Regarding the lgpu file names I have no idea.

Comment (AmintorDusko) [reply .github/workflows/tests_lqcpu_python.yml]: @maliasadi, I checked the lgpu name situation. I'm not touching these names here, but apparently, GH thinks these files still have the wrong name in this branch. I'm positive (but not 100% sure) that if we merge it to master no change will propagate. I never saw these before.

Comment (maliasadi) [reply .github/workflows/tests_lqcpu_python.yml]: Thanks for checking @AmintorDusko! How did you test the support 3 days ago though? :thinking: 
https://github.com/PennyLaneAI/pennylane-lightning/actions/workflows/compat-check-stable-latest.yml
(I'd like us to double-check if there are more issues that should be fixed)

Comment (maliasadi) [reply .github/workflows/tests_lqcpu_python.yml]: [08f9487](https://github.com/PennyLaneAI/pennylane-lightning/pull/920/commits/08f94879b39c4a5a5b4752f37f4e35375df520b9) :partying_face:

File: .github/workflows/tests_lqcpu_python.yml [hunk 6]
```diff
@@ -195,6 +195,13 @@ jobs:
           git log -1 --format='%H'
           git status
 
+      - name: Patching tests following deprecation FIXME(remove after v0.39)
+        if: inputs.lightning-version == 'stable' && inputs.pennylane-version == 'stable'
```
Comment (AmintorDusko) [inline .github/workflows/tests_lqcpu_python.yml:199]:
```suggestion
        if: inputs.lightning-version == 'stable' && inputs.pennylane-version == 'latest'
```

**Review by maliasadi — APPROVED** [review]
Nice work! :raised_hands:

**Pull request merged**